pinot-transport/src/main/java/com/linkedin/pinot/requestHandler/BrokerRequestHandler.java

```diff
@@ -641,17 +573,17 @@ private BrokerResponse processFederatedBrokerRequest(@Nonnull BrokerRequest orig
    *
    * @param tableName table name.
    * @param responseMap map from server to response.
-   * @param responseSequence response sequence for federated request.
+   * @param isOfflineTable whether the responses are from an OFFLINE table.
    * @param dataTableMap map from server to data table.
    * @param processingExceptions list of processing exceptions.
    */
   private void deserializeServerResponses(@Nonnull String tableName, @Nonnull Map<ServerInstance, ByteBuf> responseMap,
-      int responseSequence, @Nonnull Map<ServerInstance, DataTable> dataTableMap,
+      boolean isOfflineTable, @Nonnull Map<ServerInstance, DataTable> dataTableMap,
       @Nonnull List<ProcessingException> processingExceptions) {
     for (Entry<ServerInstance, ByteBuf> entry : responseMap.entrySet()) {
       ServerInstance serverInstance = entry.getKey();
-      if (responseSequence >= 0) {
-        serverInstance = new ServerInstance(serverInstance.getHostname(), serverInstance.getPort(), responseSequence);
+      if (!isOfflineTable) {
+        serverInstance = new ServerInstance(serverInstance.getHostname(), serverInstance.getPort(), 1);
```

Comment:
We need this sequence because offline and realtime table could co-locate on the same server instance. Without this sequence, one of the response will be override.